controller/src/main/java/io/pravega/controller/server/ControllerServiceMain.java

```diff
@@ -221,4 +221,14 @@ public void forceClientSessionExpiry() throws Exception {
         ((CuratorFramework) this.storeClient.getClient()).getZookeeperClient().getZooKeeper()
                                                          .getTestable().injectSessionExpiration();
     }
+
+    @Override
+    protected void shutDown() throws Exception {
```

Comment:
I understand that this is invoked from `stopAsync`? Can we unit test this flow?

Reply:
ControllerServiceMain.stop async awaitTerminated is unit tested here:

`ControllerServiceMainTest:testControllerServiceMainStartStop`

Reply:
Ok, the code coverage is complaining about the shutdown hook, but I suppose there isn't much sense in unit testing it, is there?

Reply:
added a test to ControllerServiceMain to verify that onShutdown hook there is clean shutdown of controllerServiceMain.